### PR TITLE
clean / simplify downtimes, acnowledges and comments

### PR DESCRIPTION
**Pull request by ddurieux**

### CODE REVIEW

File: alignak/acknowledge.py [hunk 1]
```diff
@@ -68,6 +69,7 @@ class Acknowledge(AlignakObject):  # pylint: disable=R0903
         'end_time': None,
         'author': None,
         'comment': None,
+        'comment_id': StringProp(default='')
```
Comment (mohierf) [inline alignak/acknowledge.py]: Why adding this parameter? And why making it a StringProp ... a string should be enough, no? Explain in the comments ...

File: alignak/external_command.py [hunk 2]
```diff
@@ -1680,8 +1670,9 @@ def del_all_host_comments(self, host):
         :type host: alignak.objects.host.Host
         :return: None
         """
-        for comment in host.comments:
-            self.del_host_comment(comment)
+        comments = host.comments.keys()
```
Comment (mohierf) [inline alignak/external_command.py]: I do not understand... host.comments is a dictionnary, right? Indexed with the comment uuid, right? So, why not keeping the previous code that should suit for this?

File: alignak/external_command.py [hunk 3]
```diff
@@ -1706,8 +1698,9 @@ def del_all_svc_comments(self, service):
         :type service: alignak.objects.service.Service
         :return: None
         """
-        for comment in service.comments:
-            self.del_svc_comment(comment)
+        comments = service.comments.keys()
+        for uuid in comments:
```
Comment (mohierf) [inline alignak/external_command.py]: Same comment as the previous one...

File: alignak/external_command.py (hunk 2 above)
Comment (ddurieux) [inline alignak/external_command.py]: because can have the error `dictionary changed size during iteration`

File: alignak/acknowledge.py (hunk 1 above)
Comment (ddurieux) [inline alignak/acknowledge.py]: to reference the comment added in the host/service when create the acknowledge, like this, when delete the acknowledge, we delete the comment.
For the StringProp, I added a string property like other alignak objects

Comment (mohierf) [inline alignak/acknowledge.py]: But a comment uuid is not a StringProp but a Python string ... as other alignak objects uuid

**Comment by ddurieux** [comment]
I tested load retention stored in backend before this patch, works fine.
The retention seems working better with this patch, I do it many times today and no problem

**Pull request merged**